safekeeper/src/metrics.rs

```diff
@@ -63,13 +66,120 @@ pub static PERSIST_CONTROL_FILE_SECONDS: Lazy<Histogram> = Lazy::new(|| {
 });
 pub static PG_IO_BYTES: Lazy<IntCounterVec> = Lazy::new(|| {
     register_int_counter_vec!(
-        "safekeeper_pg_io_bytes",
+        "safekeeper_pg_io_bytes_total",
         "Bytes read from or written to any PostgreSQL connection",
-        &["direction"]
+        &["client_az", "sk_az", "app_name", "dir", "same_az"]
```

Comment:
for me this is strange that we have both az's AND precomputed `client_az == sk_az`. Is that intentional? Surely this could be handled in promql?

Reply:
Related to cardinality of app_name; is this something like 2 (pg14, pg15) and more when we do a rolling version update, or something much more?

Reply:
> Related to cardinality of app_name

`app_name` currently has only two values for safekeeper connections: "pageserver" or "walproposer".

> Surely this could be handled in promql?

Probably, wanted to have a convenient way to make the following queries:
- `sum(safekeeper_pg_io_bytes_total{same_az="true"})`
- `sum(safekeeper_pg_io_bytes_total{same_az="false"})`
- `sum(safekeeper_pg_io_bytes_total{same_az="unknown"})`

Don't know how to do that in PromQL without `same_az`, but maybe my googling is just bad :)

Also note that additional labels shouldn't hurt, as long as there are not many different **sets** of labels. In current setup we should have only `O(number_of_availability_zones)` different metrics, because `client_az` is the only truly variable label.

Reply:
Do we event set app_name for pageserver conns? Couldn't find that quickly.
 (it would be useful anyway)

Reply:
UPD found, it is in walreceiver_connection, not in construction of connection string.

Reply:
>> Surely this could be handled in promql?
>
> Probably, wanted to have a convenient way to make the following queries:

Well, in future it will be easy to drop this if we learn it's easy.

Also seems that cardinality is low so resolving. Thanks for explaining, I probably had a wrong idea of the sets or unique label values.